### PR TITLE
Update the generic-sensor IDL file

### PR DESCRIPTION
**Pull request by lukebjerring**
Hello reviewer(s),

This PR is intended to consolidate the spec’s IDL definition with the WPT test suite’s copy, and any idlharness tests.

The up-to-date copy of the IDL file was automatically extracted from the reffy-reports repo (https://github.com/tidoust/reffy-reports/tree/master/whatwg/idl) which scrapes known specs automatically + regulary.

This PR is part of a migration project which will eventually be automatically updating and creating PRs for changes in spec IDL.

Please check that:
The spec (and its source) is correct and up-to-date
All tests which cover the IDL in the spec have been migrated to fetch + use the idl in the `interfaces/` directory (instead of inline copies in the test files).

### CODE REVIEW

File: generic-sensor/idlharness.https.window.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+// https://w3c.github.io/sensors/
+
+'use strict';
+
+function cast(i, t) {
+  return Object.assign(Object.create(t.prototype), i);
+}
+
+promise_test(async () => {
```
Comment (Honry) [inline generic-sensor/idlharness.https.window.js]: @lukebjerring, thanks for the improvement!
Can we use `idl_test` helper, it's more simplified. :)

Comment (lukebjerring) [reply generic-sensor/idlharness.https.window.js]: Done.

File: generic-sensor/idlharness.https.window.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+// https://w3c.github.io/sensors/
+
+'use strict';
+
+function cast(i, t) {
+  return Object.assign(Object.create(t.prototype), i);
+}
+
+idl_test(
+  ['generic-sensor'],
```
Comment (Honry) [inline generic-sensor/idlharness.https.window.js]: s/generic-sensor/sensors

It's file name under `interfaces/` folder is `sensors.idl`

Comment (Honry) [reply generic-sensor/idlharness.https.window.js]: Oops, I didn't see you've rename the file. :p

**Review by Honry — APPROVED** [review]
LGTM, thanks!

**Pull request merged**